### PR TITLE
refactor: add actual stub view for React DevTools panel

### PR DESCRIPTION
**Pull request by hoxyq**
# Summary

Stacked on https://github.com/facebookexperimental/rn-chrome-devtools-frontend/pull/15. Review only the [last commit at the top](https://github.com/facebookexperimental/rn-chrome-devtools-frontend/pull/17/commits/2532dfeb2d7922aed5a6830ec800014c88703f3b).

- `panels/react_devtools_placeholder` -> `panels/react_devtools`
- Placeholder view is rendered by default
- If `Enable React DevTools` experiment is enabled, the new custom view should be rendered, which is currently empty, but will soon host React DevTools frontend.

# Test plan

Without feature flag enabled in `Experiments`:
Screenshot 2024-04-09 at 17 53 38

With feature flag enabled in `Experiments`:
Screenshot 2024-04-09 at 17 54 01

# Upstreaming plan

- [ ] This commit should be sent as a patch to the upstream `devtools-frontend` repo. I've reviewed the [contribution guide](https://docs.google.com/document/d/1WNF-KqRSzPLUUfZqQG5AFeU_Ll8TfWYcJasa_XGf7ro/edit#heading=h.9kj7femz1xg5).
- [x] This commit is React Native-specific and cannot be upstreamed.

### CODE REVIEW

File: front_end/models/react_native/FuseboxBindingsModel.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+import type * as Common from '../../core/common/common.js';
+
+type JSONValue = null | string | number | boolean | {[key: string]: JSONValue} | JSONValue[];
+type DomainName = 'react-devtools';
+type DomainMessageListener = (message: JSONValue) => void;
+type BindingCalledEventTargetEvent = Common.EventTarget.EventTargetEvent<SDK.RuntimeModel.EventTypes[SDK.RuntimeModel.Events.BindingCalled]>;
+
+const RUNTIME_GLOBAL = '__FUSEBOX_REACT_DEVTOOLS_DISPATCHER__';
+
+export class FuseboxBindingsModel extends SDK.SDKModel.SDKModel {
```
Comment (huntie) [inline front_end/models/react_native/FuseboxBindingsModel.ts:16]: (Continued) nit on naming here: I think we can assume the whole of `rn-chrome-devtools-frontend` to be within the new debugging stack. This is the `BindingsModel` for the React DevTools panel. Please either prefix with, or omit, "Fusebox".

```suggestion
export class ReactDevToolsBindingsModel extends SDK.SDKModel.SDKModel {
```

**Review by huntie — APPROVED** [review]
LGTM, with feedback on naming.

File: front_end/panels/react_devtools/react_devtools-meta.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  persistence: UI.ViewManager.ViewPersistence.PERMANENT,
+  async loadView() {
+    const Module = await loadModule();
+
+    if (Root.Runtime.Runtime.isDescriptorEnabled({
+      experiment: Root.Runtime.ExperimentName.ENABLE_REACT_DEVTOOLS_PANEL,
+      condition: null,
+    })) {
+      return Module.ReactDevToolsView.ReactDevToolsViewImpl.instance();
+    }
```
Comment (hoxyq) [inline front_end/panels/react_devtools/react_devtools-meta.ts:49]: cc @huntie for future references on using experiments declaratively

Comment (motiz88) [reply front_end/panels/react_devtools/react_devtools-meta.ts:49]: @hoxyq how about creating a wiki page (GH or internal) with these learnings?

**Pull request merged**